devdocs/settingUp.md

```diff
@@ -1,12 +1,24 @@
-#Setting Up the Developer Environment
+#Table of Contents
+1. [Setting Up the Developer Environment](#1)
+2. [Prerequisites](#2)<br>
+3. [Setting up the dev server](#3)
+4. [Running the test suite](#4)
+5. [Deploying to a staging server](#5)
+6. [Running client scripts](#6)
+7. [Troubleshooting](#7)
+8. [Tool stack](#toolStack)
+
+
+
+##<a name="1"></a>Setting Up the Developer Environment
 >If you encounter any problems during the setting up process, please refer to our [troubleshooting guide](troubleshooting-guide.md) before posting a help request in our [issue tracker](https://github.com/TEAMMATES/teammates/issues).
 
 These instructions work for Linux, OS X as well as for the Windows
 environment. The only difference for Windows environment is that the command `./gradlew` should be replaced by `gradlew.bat` everywhere.
 
 The full tool stack is given at the [end of this document](#toolStack).
 
-##Prerequisites
+##<a name="2"></a>Prerequisites
```

Comment:
There is no need to add these. Refer this for an example https://github.com/TEAMMATES/teammates/edit/master/devdocs/orientation-guide.md